Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -204,15 +204,8 @@ private class TopSitesDataSource: NSObject, UICollectionViewDataSource {
         if indexPath.item < 6 {
             let cell = collectionView.dequeueReusableCellWithReuseIdentifier(ThumbnailIdentifier, forIndexPath: indexPath) as ThumbnailCell
             cell.textLabel.text = site.title.isEmpty ? site.url : site.title
-            cell.image = nil
-
-            if let url = NSURL(string: site.url) {
-                profile.thumbnails.get(url) { (thumbnail: Thumbnail?) in
-                    if let thumbnail = thumbnail {
-                        cell.image = thumbnail.image
-                    }
-                }
-            }
+            let key = SDWebThumbnails.getKey(site.url)
+            cell.imageView.sd_setImageWithURL(NSURL(string: key))
```

Comment:
SDWebImage was designed to do this. i.e. If a different image is applied to the cell while a request is processing, it will cancel the original request.